src/extra/widgets/span/lv_span.h

```diff
@@ -0,0 +1,213 @@
+/**
+ * @file lv_span.h
+ *
+ */
+
+#ifndef LV_SPAN_H
+#define LV_SPAN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+
+#define LV_USE_SPAN 1
+#if LV_USE_SPAN != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef LV_SPAN_SNIPPET_STACK_SIZE
+#define LV_SPAN_SNIPPET_STACK_SIZE 64
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+enum {
+    LV_SPAN_ALIGN_LEFT,    /**< Align text to left */
+    LV_SPAN_ALIGN_CENTER,  /**< Align text to center */
+    LV_SPAN_ALIGN_RIGHT,   /**< Align text to right */
+    LV_SPAN_ALIGN_AUTO,    /**< Use LEFT or RIGHT depending on the direction of the text (LTR/RTL)*/
+};
+typedef uint8_t lv_span_align_t;
```

Comment:
Can we use `lv_text_align_t` instead? 
In v8 it's style property (called `text_align`). I think it's reasonable to follow this pattern here too.